<?xml version="1.0"?>
<launch>
    <arg name="ns" default="local_planner"/>

    <!-- param -->
    <arg name="hz" default="20"/>
    <arg name="subscribe_count_th" default="3"/>
    <arg name="sleep_time_after_finish" default="0.5"/>
    <arg name="planner_param" default="$(find point_follow_planner)/config/planner_param.yaml"/>
    <arg name="robot_param" default="$(find point_follow_planner)/config/robot_param.yaml"/>
    <!-- topic name -->
    <arg name="cmd_vel" default="/local_path/cmd_vel"/>
    <arg name="footprint" default="/footprint"/>
    <arg name="local_goal" default="/local_goal"/>
    <arg name="local_map" default="/local_map"/>
    <arg name="odom" default="/odom"/>
    <arg name="target_velocity" default="/target_velocity"/>

    <!-- run point_follow_planner node -->
    <node pkg="point_follow_planner" type="point_follow_planner" name="point_follow_planner" ns="$(arg ns)" output="screen">
        <!-- param -->
        <rosparam command="load" file="$(arg planner_param)"/>
        <rosparam command="load" file="$(arg robot_param)"/>
        <param name="hz" value="$(arg hz)"/>
        <param name="subscribe_count_th" value="$(arg subscribe_count_th)"/>
        <param name="sleep_time_after_finish" value="$(arg sleep_time_after_finish)"/>
        <!-- topic name -->
        <remap from="/cmd_vel" to="$(arg cmd_vel)"/>
        <remap from="/footprint" to="$(arg footprint)"/>
        <remap from="/local_goal" to="$(arg local_goal)"/>
        <remap from="/local_map" to="$(arg local_map)"/>
        <remap from="/odom" to="$(arg odom)"/>
        <remap from="/target_velocity" to="$(arg target_velocity)"/>
    </node>
</launch>
